pkg/pod/status.go

```diff
@@ -143,15 +152,80 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 	return *trs, merr.ErrorOrNil()
 }
 
-func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun) *multierror.Error {
+func getPodLogs(ctx context.Context, clientset kubernetes.Interface, namespace string, name string, container string) (string, error) {
+	podLogOpts := corev1.PodLogOptions{Container: container}
+	req := clientset.CoreV1().Pods(namespace).GetLogs(name, &podLogOpts)
+	podLogs, err := req.Stream(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
```

Comment:
Something I've also been think about for the oidc way is how much do we want to load into memory at a time -

you may want to use a https://pkg.go.dev/bufio#Scanner as a first step to avoid loading the whole raw blob at once - we may also want to consider chunking the updates in a later PR as well.

Reply:
Good point! The way, I write out results in the sidecar container is line by line so I could use the Scanner's default split function which does load in one line at a time. That would read the chunks one result at a time.

Reply:
Also, the default buffer cap size is 4096 Bytes which is larger than what we expect our results to be so we may not want to change the buffer capacity either.